FoodplannerDataAccessSql/Codes/OneTimePasswordRepository.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Dapper;
+using FoodplannerModels.Account;
+using FoodplannerModels.Codes;
+using static System.Net.WebRequestMethods;
+
+namespace FoodplannerDataAccessSql.Codes
+{
+    public class OneTimePasswordRepository : IOneTimePasswordRepository
+    {
+        private readonly PostgreSQLConnectionFactory _connectionFactory;
+
+        public OneTimePasswordRepository(PostgreSQLConnectionFactory connectionFactory)
+        {
+            _connectionFactory = connectionFactory;
+        }
+
+        public async Task<bool> CheckIfCodeExistsAsync(string code)
+        {
+            const string sql = "SELECT code_id FROM one_time_password WHERE code = @Code";
+            using var connection = _connectionFactory.Create();
+            connection.Open();
+
+            var result = await connection.QuerySingleOrDefaultAsync<int?>(sql, new { Code = code });
+
+            if(result == null)
+                return false;
+            else
+                return true;
+        }
+
+        public async Task<bool> CheckIfCodeExpiredAsync(string code)
+        {
+            const string sql = "SELECT expires_on FROM one_time_password WHERE code = @Code";
+            using var connection = _connectionFactory.Create();
+            connection.Open();
+
+            var result = await connection.QuerySingleOrDefaultAsync<DateTime?>(sql, new { Code = code });
+
```

Comment:
Remember to close the connection when you are done with it:
```suggestion

            connection.Close();

```